lib/ldap_fluff/ad_member_service.rb

```diff
@@ -32,12 +32,15 @@ def _walk_group_ancestry(group_dns = [])
     group_dns.each do |group_dn|
       search = @ldap.search(:base => group_dn, :scope => Net::LDAP::SearchScope_BaseObject, :attributes => ['memberof'])
       if !search.nil? && !search.first.nil?
-        group = search.first
-        set  += _walk_group_ancestry(group[:memberof])
-        set  += group[:memberof]
+        groups = search.first[:memberof] - known_groups
```

Comment:
I'm not sure of what you mean by the existing list of groups. If you mean the `set` variable, it is empty when you recursevly re-enter`_walk_group_ancestry`.
Only by the end of all recursions will the arrays contain the same data.